src/main/java/org/breedinginsight/brapps/importer/services/BrAPIFileImportService.java

```diff
@@ -0,0 +1,289 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Singleton
+public class BrAPIFileImportService {
+
+    private ProgramUserService programUserService;
+    private ProgramService programService;
+    private MimeTypeParser mimeTypeParser;
+    private ImportMappingDAO importMappingDAO;
+    private ObjectMapper objectMapper;
+    private BrAPIMappingManager mappingManager;
+    private BrAPIImportConfigManager configManager;
+
+    @Inject
+    BrAPIFileImportService(ProgramUserService programUserService, ProgramService programService, MimeTypeParser mimeTypeParser,
+                           ImportMappingDAO importMappingDAO, ObjectMapper objectMapper, BrAPIMappingManager mappingManager,
+                           BrAPIImportConfigManager configManager) {
+        this.programUserService = programUserService;
+        this.programService = programService;
+        this.mimeTypeParser = mimeTypeParser;
+        this.importMappingDAO = importMappingDAO;
+        this.objectMapper = objectMapper;
+        this.mappingManager = mappingManager;
+        this.configManager = configManager;
+    }
+
+    public List<ImportConfig> getAllImportTypeConfigs() {
+        return configManager.getAllImportTypeConfigs();
+    }
+    /*
+        Saves the file for the mapping record
+     */
+    public BrAPIImportMapping createMapping(UUID programId, AuthenticatedUser actingUser, CompletedFileUpload file) throws
+            DoesNotExistException, AuthorizationException, HttpBadRequestException, UnsupportedTypeException {
+
+        if (!programService.exists(programId))
+        {
+            throw new DoesNotExistException("Program id does not exist");
+        }
+
+        if (!programUserService.existsAndActive(programId, actingUser.getId())) {
+            throw new AuthorizationException("User not in program");
+        }
+
+        Table df = parseUploadedFile(file);
+
+        // Convert the table to json to store
+        String jsonString = df.write().toString("json");
+        JSONB jsonb = JSONB.valueOf(jsonString);
+        ImportMappingEntity importMappingEntity = ImportMappingEntity.builder()
+                .programId(programId)
+                .file(jsonb)
+                .draft(true)
+                .createdBy(actingUser.getId())
+                .updatedBy(actingUser.getId())
+                .build();
+        importMappingDAO.insert(importMappingEntity);
+
+        BrAPIImportMapping importMapping = importMappingDAO.getMapping(importMappingEntity.getId()).get();
+
+        return importMapping;
+    }
+
+    private Table parseUploadedFile(CompletedFileUpload file) throws HttpBadRequestException, UnsupportedTypeException {
+
+        MediaType mediaType;
+        try {
+            mediaType = mimeTypeParser.getMimeType(file);
+        } catch (IOException e){
+            throw new HttpBadRequestException("Could not determine file type");
```

Comment:
HttpStatusException(HttpStatus.BAD_REQUEST, "Could not determine file type")

Reply:
pushed, there is a new build error

Reply:
fixed